From c2303dff95aa174021a1950656fdf9a1cf983959 Mon Sep 17 00:00:00 2001
From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Sat, 8 Jul 2017 09:47:12 +0100
Subject: [PATCH] aclocal.m4: allow arbitrary <vendor> string in toolchain
 triplets

Canonical triplets have a form of
    <arch>-<vendor>-<os>[-<abi>]

Checking for vendor is almost never correct as it's an
arbitrary string.

It's useful to have multiple "vendors" to denote
otherwise the same (WRT <arch>, <os>, <abi>) target:
    --target=x86_64-pc-linux-gnu
    --target=x86_64-unknown-linux-gnu
    --target=x86_64-ghc80-linux-gnu
    --target=x86_64-ghchead-linux-gnu

Do not fail unknown vendors. Only emit a warning.
Ideally configure checks should never use "vendor".

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
---
 aclocal.m4 | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/aclocal.m4 b/aclocal.m4
index 001f813dfc..1d9c09b0cd 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -230,8 +230,7 @@ AC_DEFUN([FPTOOLS_SET_HASKELL_PLATFORM_VARS],
         dec|none|unknown|hp|apple|next|sun|sgi|ibm|montavista|portbld)
             ;;
         *)
-            echo "Unknown vendor [$]1"
-            exit 1
+            AC_MSG_WARN([Unknown vendor [$]1])
             ;;
         esac
     }
-- 
2.13.3

